dynamic_listings/python/default/containers/particle_emitter_complex.py

```diff
@@ -3,17 +3,17 @@ class Emitter:
     This is a simple particle emitter, it contains a list
     of particles and it updates and manages them
     """
-    origin = None
-    particles = None
+    origin: Vector2 = None
+    particles: list[Particle] = None
     # Defines if this emitter streams continuously or only a burst of particles
     one_shot = False
 
-    def __init__(self, location, one_shot=False):
+    def __init__(self, location: Vector2, one_shot: bool = False):
         self.origin = location
-        self.particles = []  # We prepare a list of particles, we'll use 8
+        self.particles = []  # We prepare a list of particles
```

Comment:
I just realized that here a list meant to contain particles is prepared, but no particles are initialized. That's on me.
It would probably be something akin to
```
...
self.particles = [Particle() for _ in range(8)]
...
```